### PR TITLE
openai: add python3Packages.wandb dependency

### PR DESCRIPTION
**Pull request by malob**
###### Motivation for this change

The `openai` executable from `python3Packages.openai` added new functionality that depends on `python3Packages.wandb`. This PR adds that dependency, and also fixes some build issues with packages `wandb` depends on.

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [x] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by malob** [comment]
Looks like the fixes to the dependencies of `wandb` will require rebuilding over a thousand packages!!!

I'm running `nixpkgs-review` on my systems, but it's doing to take a while :)

**Comment by malob** [comment]
The @ofborg failure on Linux seems to be due to `python3Packages.sanic` (which is a dependency of `wandb` failing to build. I'm looking into it.

File: pkgs/development/python-modules/dnspython/default.nix
Comment (gador) [inline pkgs/development/python-modules/dnspython/default.nix:36]: are these changes necessary to build on Darwin?

File: pkgs/development/python-modules/dnspython/default.nix
Comment (gador) [inline pkgs/development/python-modules/dnspython/default.nix:30]: Do we need to remove this test? If I recall correctly this builds on hydra

**Review by gador — CHANGES_REQUESTED** [review]
The mass rebuild is due to changes in `pkgs/development/python-modules/dnspython/default.nix`
Many python packages depend on this package. 

Do you need to change it for openai to work? If so: Please add the changes to dnspython in a seperate PR against staging.

File: pkgs/development/python-modules/dnspython/default.nix
Comment (malob) [reply pkgs/development/python-modules/dnspython/default.nix]: Yes, at least on my machine, I get failing tests saying, e.g.:
```
ssl.SSLCertVerificationError: [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:1129)
```

I just force pushed that adds it to `checkInputs`, instead of a `nativeBuildInputs`.

Comment (malob) [reply pkgs/development/python-modules/dnspython/default.nix]: I also see the same error on hydra.

File: pkgs/development/python-modules/dnspython/default.nix
Comment (malob) [reply pkgs/development/python-modules/dnspython/default.nix:32]: You are correct that it does build on Hydra, but it won't build locally on my machine, since the network I'm on doesn't provide IPv6. So seems good to disable it in case people end up building locally.

**Comment by malob** [comment]
> Do you need to change it for openai to work? If so: Please add the changes to dnspython in a seperate PR against staging.

@gador, `dnspython` is a dependency of `wandb`. `wandb` isn't required for `openai` to build, but some functionality of the executable isn't available without it, so I'd like to add it.

Just to confirm, you'd like me to make a new PR with the changes to `dnspython`, that targets `staging`, then come back to this PR with that change removed once the new PR lands?

**Comment by malob** [comment]
Blocked on: #161740

**Comment by gador** [comment]
I'd think that's a sensible action, yes.

This way, this PR gets reviewed and merged faster and mass rebuilds (due to dnspython) should go to staging (and will take some time to get merged)

**Review by jonringer — APPROVED** [review]
LGTM

Result of nixpkgs-review pr 161725 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)<details>  <summary>1 package built:</summary>  <ul>    <li>openai (python39Packages.openai)</li>  </ul></details>

**Comment by jonringer** [comment]
to rebase to staging:

```
# finding the common merge base will avoid pinging all CODEOWNERs
common=$(git merge-base origin/master origin/staging)
git rebase --onto=$common HEAD~1 # or however many commits you want to pull
git push .. ... --force
```
then change the base branch in the github PR from `master` -> `staging`

See https://nixos.org/nixpkgs/manual/#submitting-changes-staging-branch for more details on staging branch

**Pull request merged**
